### PR TITLE
1367 geothermalheatpump example fails in openmodelica

### LINKED ISSUE #1367: GeothermalHeatPump example fails in OpenModelica
Error in model:  AixLib.Fluid.Examples.GeothermalHeatPump.GeothermalHeatPump 
Failed to build model: AixLib.Fluid.Examples.GeothermalHeatPump.GeothermalHeatPump
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:312:11-312:43:writable] Warning: Conditional component ‘mHC1_flow_nominal‘ is used in a non-connect context.
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:305:5-305:34:writable] Warning: Conditional component ‘m_flow_small_HC1‘ is used in a non-connect context.
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:333:3-334:118:writable] Warning: Conditional component ‘mHC1_flow_nominal‘ is used in a non-connect context.
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:312:11-312:43:writable] Warning: Conditional component ‘mHC1_flow_nominal‘ is used in a non-connect context.
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:305:5-305:34:writable] Warning: Conditional component ‘m_flow_small_HC1‘ is used in a non-connect context.
[D:/04_git/AixLib/AixLib/Fluid/Storage/BufferStorage.mo:333:3-334:118:writable] Warning: Conditional component ‘mHC1_flow_nominal‘ is used in a non-connect context.
Error: Too few equations, under-determined system. The model has 4628 equation(s) and 4629 variable(s).

### PR DESCRIPTION
**Pull request by HannahRomberg**
Closes #1367 

The flow direction from the cold storage to the evaporator was going in the wrong direction. The heating coil orientation was changed to down->up and the port connections were changed to allow the flow in the right direction.

### CODE REVIEW

File: AixLib/Fluid/Examples/GeothermalHeatPump/BaseClasses/GeothermalHeatPumpBase.mo [hunk 1]
```diff
@@ -283,11 +286,6 @@ equation
       Line(points={{33.275,-14.34},{33.275,-20},{58,-20}}, color={0,127,255}));
   connect(coldStorage.fluidportTop1, resistanceColdConsumerReturn.port_b)
     annotation (Line(points={{33.1,20.17},{33.1,32},{80,32}}, color={0,127,255}));
-  connect(heatPump.port_b2, coldStorage.portHC1Out) annotation (Line(points={{
```
Comment (FWuellhorst) [inline AixLib/Fluid/Examples/GeothermalHeatPump/BaseClasses/GeothermalHeatPumpBase.mo:286]: These connections seem to be the same as before?!

File: AixLib/Fluid/Examples/GeothermalHeatPump/BaseClasses/GeothermalHeatPumpBase.mo [hunk 2]
```diff
@@ -11,9 +11,11 @@ partial model GeothermalHeatPumpBase
...
     "Initial temperature of warm components";
 
-    replaceable AixLib.Fluid.Interfaces.PartialTwoPortTransport PeakLoadDevice(
-      redeclare package Medium = Medium)                                       constrainedby
-    AixLib.Fluid.Interfaces.PartialTwoPort
+  replaceable model PeakLoadDeviceModel =
+      AixLib.Fluid.Interfaces.PartialTwoPortTransport constrainedby
+    AixLib.Fluid.Interfaces.PartialTwoPortTransport(redeclare package Medium = Medium)
+    annotation(choicesAllMatching=true);
+    PeakLoadDeviceModel PeakLoadDevice
```
Comment (FWuellhorst) [inline AixLib/Fluid/Examples/GeothermalHeatPump/BaseClasses/GeothermalHeatPumpBase.mo:18]: Minor: 
A model has capital names `PeakLoadDeviceModel`. The instance of a model should be lower case, e.g. `peaLoaDev` (three letter camel-case name according to guidelines)

**Review by FWuellhorst — CHANGES_REQUESTED** [review]
Please change the name and check if the new connection is required. If the line is better now, you can leave it.

**Comment by DaJansenGit** [comment]
The previous pipelines failed due to another, not already solved, problem with the `AixLib.Controls.HVACAgentBasedControl` package. For now, I put the models on the whitelist and opened issue #1447 which needs to be solved independent from this pull request.

**Review by HannahRomberg — COMMENTED** [review]
Adressed requested changes. See comments.

**Review by FWuellhorst — APPROVED** [review]
Looks good, thanks!

**Pull request merged**